v0_5/src/user_stake.rs

```diff
@@ -62,11 +50,43 @@ pub trait UserStakeModule {
         self.events().stake_event(&caller, &payment);
 
         // create stake funds
+        let amount_to_stake = payment.clone();
         self.fund_transf_module().create_waiting(user_id, payment);
 
+        // get total unstaked
+        let total_unstaked = self.fund_view_module().get_user_stake_of_type(USER_STAKE_TOTALS_ID, FundType::UnStaked);
+        if total_unstaked == 0 {
+            return Ok(());
+        }
+        let swappable = core::cmp::min(&amount_to_stake, &total_unstaked);
+
+        // swap unStaked to deferred payment
+        sc_try!(self.fund_transf_module().swap_unstaked_to_deferred_payment(&swappable));
+
+        // swap waiting to active
+        let affected_users = self.fund_transf_module().swap_waiting_to_active(&swappable);
+        for user_id in affected_users.iter() {
+            self.rewards().compute_one_user_reward(*user_id);
+        }
+
         Ok(())
     }
 
+    /// Delegate stake to the smart contract. 
+    /// Stake is initially inactive, so does it not produce rewards.
+    #[payable]
+    #[endpoint(stake)]
+    fn stake_endpoint(&self, #[payment] payment: BigUint) -> SCResult<()> {
```

Comment:
It's completely fine if the owner calls this endpoint, right? They just get added as a normal user?

Reply:
yep :). the owner is with ID 1, and for later he must at least own 10% of the staked amount - for him to have skin in the game. owner_stake_share